web/src/features/charts/elements/GraphBackground.tsx

```diff
@@ -1,70 +1,67 @@
-/* eslint-disable react/jsx-handler-names */
-/* eslint-disable unicorn/no-null */
-/* eslint-disable react/display-name */
-import React from 'react';
+import { ScaleLinear, ScaleTime } from 'd3-scale';
+import React, { useCallback, useMemo } from 'react';
 
 import { detectHoveredDatapointIndex, noop } from '../graphUtils';
 
-const GraphBackground = React.memo(
-  ({
-    timeScale,
-    valueScale,
-    datetimes,
-    mouseMoveHandler,
-    mouseOutHandler,
-    isMobile,
-    svgNode,
-  }: any) => {
-    const [x1, x2] = timeScale.range();
-    const [y2, y1] = valueScale.range();
-    const width = x2 - x1;
-    const height = y2 - y1;
+interface GraphBackgroundProps {
+  timeScale: ScaleTime<number, number>;
+  valueScale: ScaleLinear<number, number>;
+  datetimes: Date[];
+  mouseMoveHandler?: (timeIndex: number | null, layerIndex: number | null) => void;
+  mouseOutHandler?: () => void;
+  isMobile: boolean;
+  svgNode: SVGSVGElement;
+  displayName?: string;
+}
 
-    // Mouse hover events
-    let mouseOutRectTimeout: string | number | NodeJS.Timeout | undefined;
-    const handleRectMouseMove = (event_: any) => {
-      if (mouseOutRectTimeout) {
-        clearTimeout(mouseOutRectTimeout);
-        mouseOutRectTimeout = undefined;
-      }
-      const timeIndex = detectHoveredDatapointIndex(
-        event_,
-        datetimes,
-        timeScale,
-        svgNode
-      );
+const GraphBackground = React.memo(function GraphBackground({
+  timeScale,
+  valueScale,
+  datetimes,
+  mouseMoveHandler,
+  mouseOutHandler,
+  isMobile,
+  svgNode,
+}: GraphBackgroundProps) {
+  const [x1, x2] = useMemo(() => timeScale.range(), [timeScale]);
+  const [y2, y1] = useMemo(() => valueScale.range(), [valueScale]);
+  const width = useMemo(() => x2 - x1, [x1, x2]);
+  const height = useMemo(() => y2 - y1, [y1, y2]);
```

Comment:
I'm not sure if it is worth using the useMemo here, and I think this looks a bit simpler:

`const width = x2 - x1;
   const height = y2 - y1;`

Reply:
Good point, I've updated them